functions/Invoke-DbaDbTransfer.ps1

```diff
@@ -0,0 +1,259 @@
+        [PSCredential]$DestinationSqlCredential,
+
+        [Parameter(ParameterSetName = "Default")]
+        [string]$Database,
+
+        [Parameter(ParameterSetName = "Default")]
+        [string]$DestinationDatabase = $Database,
+
+        [Parameter(ParameterSetName = "Default")]
+        [int]$BatchSize = 50000,
+
+        [Parameter(ParameterSetName = "Default")]
+        [int]$BulkCopyTimeOut = 5000,
+
+        [Parameter(ParameterSetName = "Default")]
+        [Microsoft.SqlServer.Management.Smo.ScriptingOptions]$ScriptingOption,
+
+        [Parameter(ValueFromPipeline, ParameterSetName = "Default")]
+        [Microsoft.SqlServer.Management.Smo.Transfer]$InputObject,
+
+        [Parameter(ParameterSetName = "Default")]
+        [switch]$CopyAllObjects,
+
+        [Parameter(ParameterSetName = "Default")]
+        [ValidateSet(
+            'FullTextCatalogs',
+            'FullTextStopLists',
+            'SearchPropertyLists',
+            'Tables',
+            'Views',
+            'StoredProcedures',
+            'UserDefinedFunctions',
+            'UserDefinedDataTypes',
+            'UserDefinedTableTypes',
+            'PlanGuides',
+            'Rules',
+            'Defaults',
+            'Users',
+            'Roles',
+            'PartitionSchemes',
+            'PartitionFunctions',
+            'XmlSchemaCollections',
+            'SqlAssemblies',
+            'UserDefinedAggregates',
+            'UserDefinedTypes',
+            'Schemas',
+            'Synonyms',
+            'Sequences',
+            'DatabaseTriggers',
+            'DatabaseScopedCredentials',
+            'ExternalFileFormats',
+            'ExternalDataSources',
+            'Logins',
+            'ExternalLibraries'
+        )]
+        [string[]]$CopyAll,
+
+        [Parameter(ParameterSetName = "Default")]
+        [switch]$SchemaOnly,
+
+        [Parameter(ParameterSetName = "Default")]
+        [switch]$DataOnly,
+
+        [switch]$ScriptOnly,
+
+        [switch]$EnableException
+    )
+    begin {
+        $newTransferParams = (Get-Command New-DbaDbTransfer).Parameters.Keys | Where-Object { $_ -notin [System.Management.Automation.PSCmdlet]::CommonParameters }
+    }
+    process {
+        if ($InputObject) {
+            $transfer = $InputObject
+        } else {
+            $paramSet = @{ }
+            # generate transfer object by adding all applicable parameters to the New-DbaDbTransfer call
+            foreach ($key in $PSBoundParameters.Keys) {
+                if ($key -in $newTransferParams) {
+                    $paramSet[$key] = $PSBoundParameters[$key]
+                }
+            }
+            Write-Message -Message "Generating a transfer object based on current parameters" -Level Verbose
+            $transfer = New-DbaDbTransfer @paramSet
+        }
+        # add event handling
+        $events = Register-ObjectEvent -InputObject $transfer -EventName DataTransferEvent -Action {
```

Comment:
This can't be used if we are to continue supporting PowerShell 3. `Register-ObjectEvent` is a Windows PowerShell 5.1 only command. // @potatoqualitee

Reply:
I'm about 80% sure...don't have Windows PowerShell 3 machine handy.

Reply:
Dacpac cmds already use that. I can add a version check around the call though once the dependency is confirmed.

Reply:
This command is indeed available in PS3, please feel free to proceed @nvarscar